lib/ex_vatcheck/vies_client.ex

```diff
@@ -42,7 +42,7 @@ defmodule ExVatcheck.VIESClient do
   def new() do
     with {:ok, response} <- HTTPoison.get(@wsdl_url),
          {:ok, url} <- XMLParser.parse_service(response.body) do
-      {:ok, %__MODULE__{url: url}}
+      {:ok, %__MODULE__{url: corrected_client_url(url)}}
```

Comment:
Not a blocker but might consider sending a parsed URI then have a sensible action if we dont get a valid URL at all.
Then can move it up into the with block

 untested code inc
`URI.parse(url)
defp correct_client_url(%{scheme: "https"} = uri), do: {:ok, URI.to_string(uri)}
defp correct_client_url(%{scheme: "http"} = uri), do: {:ok ,uri |> Map.put(:scheme, "https") |> URI.to_string()}
defp correct_client_url(_), do: {:error, "malformed url"}
`

Reply:
This is a great idea. I'm sorry I didn't notice it before merging.